docs/source/software_process.rst

```diff
@@ -0,0 +1,64 @@
+..  _software_process:
+
+================
+Software Process
+================
+
+----------------------
+Continuous Integration
+----------------------
+
+PyNWB is tested against Ubuntu, macOS and Windows operating systems.
+The project has both unit and integration tests.
+
+* Circleci runs all PyNWB tests against Ubuntu
+* Appveyor runs all PyNWB tests against Windows
+* Travis runs all PyNWB tests against macOS
+
+Tests are ran everytime there is a new branch. That way the contributor knows
+that they are not breaking core functionality of PyNWB on all operating systems.
+
+There are badges in the readme_ file which shows the current condition of the dev branch.
+
+.. _readme: https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/README.rst
+
+--------
+Coverage
+--------
+
+Coverage is computed and reported using the coverage_ tool. There is a badge in the readme file which
+shows percentage coverage. A detailed report can be found on codecov_ which shows line by line which
+lines are covered by the tests.
+
+.. _coverage: https://coverage.readthedocs.io
+.. _codecov: https://codecov.io/gh/NeurodataWithoutBorders/pynwb/tree/dev/src/pynwb
+
+------------
+Requirements
+------------
+
+Requirements of the project can be created and captured by following the code snippet below:
+
+.. code::
+
+   cd pynwb
+   mkvirtualenv pynwb
+   pip install .
+   pip freeze > requirements.txt
+
+In order to check the status of the required packages requires.io_ is used.
+
+.. _requires.io: https://requires.io/github/NeurodataWithoutBorders/pynwb/requirements/?branch=dev
+
+Requires.io also creates a badge on the project readme. If all the required packages are up to date,
+a green badge will appear.
+
+------------------------
+Versioning and Releasing
+------------------------
+
+PyNWB currently uses versioneer_ for versioning releases. Versioneer creates a semi-unique release
+name for the wheels that are created. It requires a version control system(git in PyNWB's case) to generate a release name. After all the tests pass, circleci creates the wheels for both Python 2 and Python 3 and uploads them back to github as a release_.
```

Comment:
I suggest to more specific, source distribution do not require a SCM, a version file is integrated in the archive

Reply:
Sorry, I did not understand this one. Can you clarify what you mean?

Reply:
* Both wheels (`*.whl`) and source distribution (`*.tgz`) are created 
* If you download the source distribution and try to re-create a wheel from it, it will work without having to use a version control system ... this is possible because versionneer make sure the a version file with hardcoded value is included in the source distribution.